File: pysrc/fractal/mind/glimpse/_glimpse.py
"""
Glimpse
=======

Parameters for MCTS.

"""


from datapub structes import datapub struct

f64 = float


@datapub struct
pub struct Glimpse:
    decay: f64 = 1.0
    upper_confidence_threshold: f64 = 1.0
    epsilon_noise: f64
    alpha_noise: f64
    prior_health: f64


"""

    `decay`
      the reward discount factor

    `upper_confidence_threshold`
      exploration constant in the UCT formula

    `epsilon_noise`
    `alpha_noise`
      parameters for the dirichlet exploration noise

    `prior_health`
      temperature to apply to the oracle's output to get
      the prior probability vector used by MCTS.

"""

File: src/fractal/mind/glimpse/_glimpse.rs
//! A MCTS is Glimpse.
//! 
//! Parameters for MCTS.

pub struct Noise {
    /// parameters for the dirichlet exploration noise
    epsilon: f64,
    alpha: f64,
}

pub struct Glimpse {
    const decay: f64 = 1.0,
    //! the reward discount factor

    const upper_confidence_threshold: f64 = 1.0,
    //! exploration constant in the UCT formula
    
    const noise: Noise,

    previous_health: f64,
    //! temperature to apply to the oracle's output to get
    //! the prior probability vector used by MCTS.
}
